bazaar/interface.rst

```diff
@@ -1,12 +1,207 @@
+Param                       Description                                      Type    Required
+==========================  ================================================ ======= =========
+first_name                  First name of user.                              alpha   required
+last_name                   Last name of user.                               alpha   required
+email                       Email of user.                                   email   optional
+user_id                     User id from                                     alpha   required
+context_id                  Course or page identifier. Must be unique.       alpha   required
+context_title               Course or page title                             alpha   required
+role                        Role (enum student, teacher, admin)              alpha   required
+school                      Name of the school                               alpha   required
+school_id                   School's identifier in the national level.       alpha   required
+city                        City's name                                      alpha   required
+city_id                     City's unique identifier in the national level   alpha   required
+oid                         User oid                                         alpha   optional
+add_resource_callback_url   LMS callback URL, for adding material to the LMS url     optional
+cancel_url                  Forwarding user back to  LMS                     url     optional
+==========================  ================================================ ======= =========
+
+
+1.1 Material drag and drop
+--------------------------
+
+LMS should implement drop functionality directly and parse links with "edustore.fi/<base64 encoded json>.
+Base64 encoded JSON has basic information about the resource: name, description and uid.
+
+
+2 Viewing material 
+===================
+
+Use case:
+Person wants to view selected material/resource. 
+
+API workflow:
+
+1. LMS makes a JSON POST request to the Edustore with params from table 2, using header-based auth.
+2. Edustore replies with unique url and LMS forwards user to the unique url.  
+3. User is forwarded to the resource(if resource is internal pdf, videos, epub, scorms) or forwarded to the publisher system. Publisher should implement 2.2 part of the specs.
+
+
+Edustore integrated player?
+
+1. Resource is located at the Edustore server(pdf, epub, scorm) -> resource is shown using integrated player
+2. Resource is located at publishers server step 2.2 			-> user is forwarded to the publishers server
+
+2.1 LMS to Edustore
+--------------------
+
+Workflow:
+
+1. LMS makes a JSON POST request to the Edustore with params from table 2, using header-base authentication.
+2. Edustore replies with unique url and LMS forwards user to the unique url.  
+3. Edustore displays or forwards user to the selected material.
+
+Edustore does all the license and logging functionality.
```

Comment:
This is specific to the implementation you guys have, not what might be in some other service implementing this API. This kind of language should be in the Bazaar general explanation page, not in this document.